include/swift/AST/TypeCheckRequests.h

```diff
@@ -2665,6 +2666,26 @@ class DefaultArgumentExprRequest
   void cacheResult(Expr *expr) const;
 };
 
+/// Computes the type of the default expression for a given parameter.
+class DefaultArgumentTypeRequest
+    : public SimpleRequest<DefaultArgumentTypeRequest, Type(ParamDecl *),
+                           RequestFlags::SeparatelyCached> {
```

Comment:
Out of curiosity, what's the reason for separately caching the result?

Reply:
It allows to cache the type during deserialization without having to run the request.

Reply:
Ah, you can still do that with `evaluator.cacheOutput`. Not saying you need to change this though

Reply:
Yeah, I know about that one, just seemed like a consistent thing to do since I'm not really sure which one is better and/or preferred here.